test/upgrading/upgrader.spec.js

```diff
@@ -116,6 +116,26 @@ describe('Upgrader', () => {
     expect(result).to.eql([hello])
   })
 
+  it('should upgrade with only crypto', async () => {
+    const { inbound, outbound } = mockMultiaddrConnPair({ addrs, remotePeer })
+
+    // No available muxers
+    const muxers = new Map()
+    sinon.stub(localUpgrader, 'muxers').value(muxers)
+    sinon.stub(remoteUpgrader, 'muxers').value(muxers)
+
+    const cryptos = new Map([[Crypto.protocol, Crypto]])
+    sinon.stub(localUpgrader, 'cryptos').value(cryptos)
+    sinon.stub(remoteUpgrader, 'cryptos').value(cryptos)
+
+    const connections = await Promise.all([
+      localUpgrader.upgradeOutbound(outbound),
+      remoteUpgrader.upgradeInbound(inbound)
+    ])
+
+    expect(connections).to.have.length(2)
+  })
```

Comment:
Should I test the errors with multiplexer operations in the `connection`?

Reply:
It might be good to add a quick test. They should also throw errors with codes.

Reply:
I think it would be good to also add a test for closing. It's going to the MultiaddrConn so it should be fine, but I think having it for avoid regressions in the future would be good.

Reply:
I added to this test. I think it is better than create a new test